packages/flutter/test/services/text_input_test.dart

```diff
@@ -406,6 +406,31 @@ void main() {
       expect(client.latestMethodCall, 'connectionClosed');
     });
 
+    test('TextInputClient commitContent method is called', () async {
+      final FakeTextInputClient client = FakeTextInputClient(TextEditingValue.empty);
+      const TextInputConfiguration configuration = TextInputConfiguration();
+      TextInput.attach(client, configuration);
+
+      expect(client.latestMethodCall, isEmpty);
+
+      // Send commitContent message with fake GIF data.
+      final ByteData? messageBytes = const JSONMessageCodec().encodeMessage(<String, dynamic>{
+        'args': <dynamic>[
+          1,
+          'TextInputAction.commitContent',
+          jsonDecode('{"mimeType": "image/gif", "data": "010101000", "uri": "content://com.google.android.inputmethod.latin.fileprovider/test.gif"}'),
+        ],
+        'method': 'TextInputClient.performAction',
+      });
+      await ServicesBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
+        'flutter/textinput',
+        messageBytes,
+            (ByteData? _) {},
+      );
+
+      expect(client.latestMethodCall, 'commitContent');
+    });
+
```

Comment:
Nit: For completeness, could you also add a test for this in editable_text_test.dart?  I'm thinking you could pump an app with an EditableText with an onContentCommitted callback.  Send the `TextInputAction.commitContent` message similar to how you do here in this test, then expect that some `called` boolean was set to true in onContentCommitted or something like that.

I think it would be somewhat similar to this test: https://github.com/flutter/flutter/blob/8815f60a5aa5acbd538f1489c895310524beb2a8/packages/flutter/test/widgets/editable_text_test.dart#L425

Reply:
Added